README.md

```diff
@@ -108,43 +108,94 @@ See [CONFIG.md](CONFIG.md) for details.
 See [Code Samples](https://github.com/hazelcast/hazelcast-nodejs-client/tree/master/code_samples)
 
 
-# Serialization Considerations
+# Serialization
 
-Hazelcast needs to serialize objects in order to be able to keep them in the server memory. For primitive types, it uses Hazelcast native serialization. For other complex types (e.g. JS objects), it uses JSON serialization.
+Serialization is the process of converting an object into a stream of bytes to store the object in memory, a file or database, or transmit it through network. Its main purpose is to save the state of an object in order to be able to recreate it when needed. The reverse process is called deserialization.
 
-For example, when you try to query your data using predicates, this querying is handled on the server side so Hazelcast does not have to bring all data to the client but only the relevant entries. Otherwise, there would be a lot of unneccessary data traffic between the client and the server and the performance would severely drop.
-Because predicates run on the server side, the server should be able to reason about your objects. That is why you need to implement serialization on the server side.
+> **NOTE: Hazelcast Node.js Client is a TypeScript-based project but JavaScript does not have interfaces. Therefore, 
+ some interfaces are given to user by using the TypeScript files that have `.d.ts` extension. In the documentation, implementing an interface
+ means an object to have the necessary functions that are listed in the interface inside the `.d.ts` file. Also, this object is mentioned as 
+ `an instance of the interface`.**
 
-The same applies to MapStore. The server should be able to deserialize your objects in order to store them in MapStore.
+Hazelcast serializes all your objects into an instance of `Data`. `Data` is the binary representation of an object. When Hazelcast serializes an object into `Data`, it first checks whether the object is an instance of `IdentifiedDataSerializable`, if not it checks if it is an instance of `Portable` and serializes it accordingly. For the following types Hazelcast optimizes the serialization and you cannot override this behavior.
```

Comment:
I wrote this before but looks like the comment is lost. `Data` is an implementation detail. There is no need for the user to know about it.

What the user needs to know in the beginning is (preferably in this order):
1. Hazelcast has a serialization method for some well known types
    - types and their Java counterparts as explained below is very nice.
2. The client JSON serializes anything that is not Hazelcast native serializable by default.
    - It is important to emphasize that this behaviour can be overridden by using one of the following methods.
3. Hazelcast has some ways for users to plug-in their own serialization code
     1. IdentifiedDataSerializable
    2. Portable
    3. Custom Serializer
    4. Global Serializer

First you told the user that he may want to know about something called serialization above. This is good. Now tell the user there is a default way of handling things in Hazelcast Node.js Client. Tell them why they might not like the default behavior sometimes (language interoperability, entry processors). Tell them they have some ways of overriding the default behavior. Give them an outline of how many ways they have to override the behavior and their names. Then explain all of them one by one.

Do not try to tell many different things in one sentence (even in one paragraph). Above paragraph is trying to tell 3 different things in 2 sentences. Let me tell you what it tries to tell
- There is a default serialization
- There are ways to override default serialization
- The order of non-default serialization preference

Each of these should be in their own simple parahgraphs.